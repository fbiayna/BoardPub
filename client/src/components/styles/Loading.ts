import { StyleSheet } from 'react-native'

const style = StyleSheet.create({
  loadingImageContainer: {
    flex: 1,
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center'
  },
  loadingImage: {
    height: 200,
    width: 200
  }
})

export default style
