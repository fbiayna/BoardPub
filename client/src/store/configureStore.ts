import { applyMiddleware, compose, createStore } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from '../reducers/boardPubReducer'

export default function configureStore (initialState: {} | undefined) {
  return createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
}
