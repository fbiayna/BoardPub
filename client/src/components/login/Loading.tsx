/* eslint-disable no-unused-expressions */
/* eslint-disable no-use-before-define */
import React, { useEffect } from 'react'
import { ImageBackground, View, Image } from 'react-native'
import styles from '../styles/loginLoadingStyles'
import { logoBoardPub, loginBackground } from '../../utils/images'
import { useIsFocused } from '@react-navigation/native'
import { connect } from 'react-redux'
import { checkIfLoggedIn } from '../../actions/authFunctions'
import { authReducer } from 'utils/interfaces'

function Loading ({ dispatch, logInExists, navigation }: authReducer) {
  const isFocused = useIsFocused()
  if (isFocused) {
    setTimeout(function () {
      dispatch(checkIfLoggedIn())
    }, 2500)
  }

  useEffect(() => {
    logInExists !== undefined
      ? logInExists
          ? navigation.reset({
              index: 0,
              routes: [{ name: 'application' }]
            })
          : navigation.reset({
            index: 0,
            routes: [{ name: 'loginStart' }]
          })
      : null
  }, [logInExists])

  return (
    <View style={styles.container} testID="loading">
      <ImageBackground source={loginBackground()} style={styles.backimage}>
        <View style={styles.shadow}>
          <View style={styles.logo}>
            <Image source={logoBoardPub()} style={styles.image} />
          </View>
        </View>
      </ImageBackground>
    </View>
  )
}

function mapStateToProps ({ userReducer, authReducer }: any) {
  return {
    user: userReducer.user,
    logInExists: authReducer.logInExists
  }
}
export default connect(mapStateToProps)(Loading)
