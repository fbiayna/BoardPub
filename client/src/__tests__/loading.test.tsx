/* eslint-disable no-use-before-define */
import { Provider } from 'react-redux'
import configureStore from '../store/configureStore'
import React from 'react'
import Loading from '../components/loading/LoadingGif'
import renderer from 'react-test-renderer'

describe('Loading Component', () => {
  test('should be defined', () => {
    const store = configureStore({})
    store.dispatch = jest.fn()

    const loading = renderer.create(<Provider store={store} >
      <Loading /></Provider>)

    expect(loading).toMatchSnapshot()
  })
})
