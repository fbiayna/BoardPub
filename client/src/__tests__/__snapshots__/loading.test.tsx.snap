// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should be defined 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://trello-attachments.s3.amazonaws.com/5ec457af3dca9f5428a9f179/210x374/aba40d5ffeb52b3299e1abb01132a636/BoardPub_-_Loading.gif",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
</View>
`;
